plugins/eos/gs-plugin-eos.c

```diff
@@ -0,0 +1,94 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2016-2018 Endless Mobile, Inc
+ *
+ * Authors:
+ *   Joaquim Rocha <jrocha@endlessm.com>
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <config.h>
+
+#include <gnome-software.h>
+#include <glib/gi18n.h>
+#include <gs-plugin.h>
+
+#include "gs-plugin-eos.h"
+
+/*
+ * SECTION:
+ * Plugin to improve GNOME Software integration in the EOS desktop.
+ *
+ * This plugin runs entirely in the main thread.
+ */
+
+struct _GsPluginEos
+{
+	GsPlugin parent;
+};
+
+G_DEFINE_TYPE (GsPluginEos, gs_plugin_eos, GS_TYPE_PLUGIN)
+
+static void
+gs_plugin_eos_init (GsPluginEos *self)
+{
+	/* let the flatpak plugin run first so we deal with the apps
+	 * in a more complete/refined state */
+	gs_plugin_add_rule (GS_PLUGIN (self), GS_PLUGIN_RULE_RUN_AFTER, "flatpak");
+}
+
+static gboolean
+app_is_flatpak (GsApp *app)
+{
+	return gs_app_get_bundle_kind (app) == AS_BUNDLE_KIND_FLATPAK;
+}
+
+void
+gs_plugin_adopt_app (GsPlugin *plugin, GsApp *app)
+{
+	if (app_is_flatpak (app))
+		return;
+
+	gs_app_set_management_plugin (app, plugin);
+}
+
+gboolean
+gs_plugin_launch (GsPlugin *plugin,
+		  GsApp *app,
+		  GCancellable *cancellable,
+		  GError **error)
+{
+	/* if the app is one of the system ones, we simply launch it through the
+	 * plugin's app launcher */
```

Comment:
I'm curious what this plugin actually does. If we remove it, can we still launch built-in apps from their page in GNOME Software?